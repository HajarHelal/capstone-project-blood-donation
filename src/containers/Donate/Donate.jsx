import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { addDonor } from '../../Utilities/FirebaseUtilities';
import FirstStep from '../../components/FirstStep';
import SecondStep from '../../components/SecondStep';
import ThirdStep from '../../components/ThirdStep';
import FourthStep from '../../components/FourthStep';

function Donate() {
  const data = {
    name: '',
    email: '',
    phone: '',
    date: '',
    weight: '',
    sickness: '',
    bloodType: '',
    gender: '',
    city: '',
  };
  const { t, i18n } = useTranslation();
  const [donor, setDonor] = useState(data);
  const [step, setstep] = useState(1);
  const [city, setcity] = useState('');
  const onChangeHandling = (e) => {
    const { name, value } = e.target;
    // if the languge is arabic and the name is of the property is city then take the arabic word
    if (i18n.language === 'ar' && name === 'city') {
      const cities = t('cities', { returnObjects: true });
      const index = cities.indexOf(value);
      const citiesEnglish = t('cities', { lng: 'en', returnObjects: true });
      setDonor({
        ...donor,
        [name]: citiesEnglish[index],
      });
    } else {
      setDonor({
        ...donor,
        [name]: value,
      });
    }
    setcity(value);
  };
  const onClickHandling = () => {
    addDonor(donor);
    setstep(4);
  };
  const chooseBlood = (value) => {
    setDonor({
      ...donor,
      bloodType: value,
    });
  };
  return (
    <div className=" py-16 sm:py-20 md:py-24 w-11/12 sm:w-10/12 mx-auto">
      {step === 1 ? (
        <FirstStep donor={donor} onChange={onChangeHandling} setStep={setstep} />
      ) : null}
      {step === 2 ? (
        <SecondStep
          donor={donor}
          onChange={onChangeHandling}
          setStep={setstep}
          chooseBlood={chooseBlood}
          city={city}
        />
      ) : null}
      {step === 3 ? (
        <ThirdStep
          donor={donor}
          onChange={onChangeHandling}
          setStep={setstep}
          onRegister={onClickHandling}
        />
      ) : null}
      {step === 4 ? <FourthStep /> : null}
    </div>
  );
}

export default Donate;
