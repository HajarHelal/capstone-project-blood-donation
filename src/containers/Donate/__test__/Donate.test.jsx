import React from 'react';
import renderer from 'react-test-renderer';
import Donate from '../Donate';

// tests will use to compare to.
it('renders correctly', () => {
  const tree = renderer.create(<Donate />).toJSON();
  expect(tree).toMatchSnapshot();
});
