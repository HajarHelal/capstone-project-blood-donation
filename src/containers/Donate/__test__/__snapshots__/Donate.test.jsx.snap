// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=" py-16 sm:py-20 md:py-24 w-11/12 sm:w-10/12 mx-auto"
>
  <div>
    <div
      className=" w-11/12 xl:w-10/12 mb-6 sm:mb-8 md:mb-12"
    >
      <div>
        <h1
          className="text-primary-100  sm:text-xl md:text-2xl lg:text-3xl pt-1 font-semibold  "
        >
          donate_first_page.title
        </h1>
        <p
          className=" pt-3 pb-5  sm:text-sm md:text-base lg:text-lg   "
        >
          donate_first_page.paragrph
        </p>
      </div>
    </div>
    <div
      className="p-6 sm:p-8 md:p-10 border-gray-300 shadow-xl border-2 rounded"
    >
      <div
        className="flex  items-center w-10/12  mb-10"
      >
        <div
          className="text-base md:text-lg lg:text-xl font-semibold  mx-4 sm:mx-8"
        >
          donate_first_page.step
          1 / 3
        </div>
        <div
          className="flex-1 h-6 sm:h-8 md:h-10 bg-gray-100"
        >
          <div
            className="w-4/12 h-6 sm:h-8 md:h-10  bg-primary"
          />
          <div
            className="w-8/12 h-6 sm:h-8 md:h-10 hidden bg-primary"
          />
          <div
            className="w-100 h-6 sm:h-8 md:h-10 hidden bg-primary"
          />
        </div>
      </div>
      <div
        className="text-base md:text-lg lg:text-xl font-medium mb-6"
      >
        donate_first_page.subHeader1
      </div>
      <div
        className="text-sm sm:text-base md:text-lg  mb-10 leading-relaxed w-10/12"
      >
        donate_first_page.subparagrph
      </div>
      <div
        className="text-base md:text-lg lg:text-xl font-medium mb-6"
      >
        donate_first_page.subHeader2
      </div>
      <div
        className=" flex flex-col items-start mb-8"
      >
        <label
          className="mb-2 tracking-normal font-medium"
        >
          donate_first_page.birth_text
        </label>
        <input
          className=" mt-4 py-2 px-4 text-sm sm:text-base tracking-normal   border-2 border-gray-300 focus:outline-none bg-white placeholder-gray-800"
          id="date"
          name="date"
          onChange={[Function]}
          type="date"
          value=""
        />
      </div>
      <div
        className="block mb-8"
        name="weight"
        onChange={[Function]}
        value=""
      >
        <span
          className="text-sm sm:text-base md:text-lg font-medium inline-block mb-2"
        >
          donate_first_page.radio_text1
        </span>
        <div
          className="mt-2"
        >
          <label
            className="inline-flex items-center"
            htmlFor="sa"
          >
            <input
              className="form-radio text-green-500 h-6 w-6"
              defaultChecked={false}
              name="weight"
              type="radio"
              value="donate_first_page.radio_option_yes"
            />
            <span
              className="mx-4"
            >
              donate_first_page.radio_option_yes
            </span>
          </label>
        </div>
        <div
          className="mt-2"
        >
          <label
            className="inline-flex items-center"
            htmlFor="sa"
          >
            <input
              className="form-radio text-green-500 h-6 w-6"
              defaultChecked={false}
              name="weight"
              type="radio"
              value="donate_first_page.radio_option_no"
            />
            <span
              className="mx-4"
            >
              donate_first_page.radio_option_no
            </span>
          </label>
        </div>
      </div>
      <div
        className="block mb-8"
        name="sickness"
        onChange={[Function]}
        value=""
      >
        <span
          className="text-sm sm:text-base md:text-lg font-medium inline-block mb-2"
        >
          donate_first_page.radio_text2
        </span>
        <div
          className="mt-2"
        >
          <label
            className="inline-flex items-center"
            htmlFor="sa"
          >
            <input
              className="form-radio text-green-500 h-6 w-6"
              defaultChecked={false}
              name="sickness"
              type="radio"
              value="donate_first_page.radio_option_yes"
            />
            <span
              className="mx-4"
            >
              donate_first_page.radio_option_yes
            </span>
          </label>
        </div>
        <div
          className="mt-2"
        >
          <label
            className="inline-flex items-center"
            htmlFor="sa"
          >
            <input
              className="form-radio text-green-500 h-6 w-6"
              defaultChecked={false}
              name="sickness"
              type="radio"
              value="donate_first_page.radio_option_no"
            />
            <span
              className="mx-4"
            >
              donate_first_page.radio_option_no
            </span>
          </label>
        </div>
      </div>
      <button
        className="inline-block text-white text-base md:text-lg  text-center md:font-medium bg-primary  md:mt-8 mt-4 px-8 md:px-12 lg:px-16 py-1 lg:py-2  shadow-md transition duration-500 ease-in-out transform hover:scale-95"
        onClick={[Function]}
        type="button"
      >
        donate_first_page.button
      </button>
    </div>
  </div>
</div>
`;
